### ISSUE TITLE
Check compatibility with MailPoet3 plugin 

### ISSUE THREAD

**Issue by sudar** [sudar/email-log#279]

**Comment by esudalaimadan** [comment]
No, email is not getting triggered in my local.  It is also not logged in email logs.  Please refer the error message under newsletter tab.

![Screenshot from 2020-06-18 18-41-54](https://user-images.githubusercontent.com/38071735/85103216-dd065280-b223-11ea-86e8-d92fab0d939b.png)

**Comment by esudalaimadan** [comment]
They are extending PHPMailer to create their own mailer class.
`class WordPressMailer extends \PHPMailer {`

**Comment by sudar** [comment]
@esudalaimadan 

Okay. See if they have any filter/hook that we can use to log the emails that are sent by Mail Poet.

**Comment by sudar** [comment]
MailPoet 3 is not firing any hooks before they send emails and therefore it is not straightforward to log emails send through MailPoet.

We may have to explore this more to find out a proper way to log emails from MailPoet 3.

See #29 regarding MailPoet 2

**Comment by BrianHenryIE** [comment]
This works:

```php
if( ! class_exists( \PHPMailer::class )) {

	class Intercept extends PHPMailer\PHPMailer {
		public function send() {

			error_log( $this->Body );

			return parent::send();
		}
	}

	/**
	 * MailPoet checks has PHPMailer already been loaded. By loading it earlier, we get to specify the class.
	 *
	 * @see PHPMailerLoader::load()
	 * @see wp-content/plugins/mailpoet/lib/Mailer/WordPress/PHPMailerLoader.php
	 */

	class_alias( Intercept::class, 'PHPMailer' );
}
```

It should be possible to hook that to only MailPoet actions, i.e. hook early on the AJAX and Cron actions that MailPoet uses.
